sound/soc/sof/topology.c

```diff
@@ -1999,7 +1999,15 @@ static int sof_route_load(struct snd_soc_component *scomp, int index,
 			goto err;
 		}
 
-		sroute->route = route;
+		sroute->route.source = devm_kstrdup(sdev->dev,
```

Comment:
devm_ should only be used for allocating resources that will be freed at remove(). This cant be used here as topology can be unloaded and then reloaded before remove(). i.e. topology is a runtime resource that will need to be manually managed. 
This code still does not check the result of the allocations.

Reply:
@lgirdwood 
Sorry for missing the check.
If we have to free it manually, it should be in the snd_sof_free_topology().
I think other project will not do it like us. 
In some code, the route is defined in static table, they did not need this kind of free, 
because they did not do it like us to attach the route in the list.

If we allocate the buffer in sof_route_load() and need to free manually in future.
I do not think we could use soc_tplg{}->ops->dapm_route_unload() to free,
although we allocate this in soc_tplg{}->ops->dapm_route_load() function.
because the input parameter of this function we can not used in free stage.
the soc_tplg{} instance is free already after the topology is finished parsing.

Do you  have any idea about this?